app/services/join_course.rb

```diff
@@ -5,11 +5,13 @@
 #= Adds a user to a course
 class JoinCourse
   attr_reader :result
+  attr_accessor :real_name
```

Comment:
We don't expect to read or set :real_name except via JoinCourse.new, I suggest leaving this out. (You may have added it to facilitate testing, but it's not necessary.)